src/gas_metering/mod.rs

```diff
@@ -468,6 +489,37 @@ fn add_grow_counter<R: Rules>(
 	b.build()
 }
 
+fn add_gas_counter(module: elements::Module, gas_global: u32) -> elements::Module {
+	use parity_wasm::elements::Instruction::*;
+
+	let mut b = builder::from_module(module);
+	b.push_function(
+		builder::function()
+			.signature()
+			.with_param(ValueType::I32)
+			.build()
+			.body()
+			.with_instructions(elements::Instructions::new(vec![
+				GetGlobal(gas_global), // (oldgas)
+				GetLocal(0),           // (oldgas) (used)
+				I32Sub,                // (newgas)
+				TeeLocal(0),           // (newgas)
+				I32Const(0),           // (newgas) (0)
+				I32LtS,                // (newgas ltz)
+				If(BlockType::NoResult),
+				Unreachable,
```

Comment:
We should either:

1. Write back the gas value here so that we know we went over. Otherwise, there's now to distinguish between this and some other crash.
2. Call a host "out of gas" function? Likely an even better approach, but I'm not sure.

Reply:
We store updated gas before getting here now